test/DynamoCoreWpfTests/UnitsUITests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Xml;
+using Dynamo.Graph;
+using Dynamo.Models;
+using Dynamo.Tests;
+using Dynamo.Utilities;
+using DynamoCoreWpfTests.Utility;
+using NUnit.Framework;
+using TestUINodes;
+using UnitsUI;
+
+namespace DynamoCoreWpfTests
+{
+    [TestFixture]
+    public class UnitsUITests : DynamoTestUIBase
+    {
+        private LengthFromString lengthFromString;
+        private MeasurementInputBaseConcrete measurementInputBase;
+        private AssemblyHelper assemblyHelper;
+
+        public override void Open(string path)
+        {
+            base.Open(path);
+
+            DispatcherUtil.DoEvents();
+        }
+
+        [SetUp]
+        public void TestsSetup()
+        {
+            lengthFromString = new LengthFromString();
+            measurementInputBase = new MeasurementInputBaseConcrete();
+        }
+
+        [Test]
+        public void MigrateLengthFromFeetToMetersTest()
+        {
+            var testDocument = new XmlDocument();
+            XmlElement root = testDocument.CreateElement("Main");
+            XmlElement systemDouble = testDocument.CreateElement("System.Double");
+            
+            //Value is 1 feet
+            systemDouble.SetAttribute("value", "1");
+            root.AppendChild(systemDouble);
+
+            lengthFromString.MigrateLengthFromFeetToMeters(root);
+
+            string oneFeetInMeters = "0.304800000001219";
+
+            Assert.AreEqual(oneFeetInMeters, systemDouble.Attributes[0].Value);
+        }
+
+        [Test]
+        public void SerializeCoreTest()
+        {
+            //Loads a value to check
+            measurementInputBase.Value = 20;
+
+            //Serializes node into xml
+            var testDocument = new XmlDocument();
+            testDocument.LoadXml("<ElementTag/>");
+            var testElement = testDocument.DocumentElement;
+            measurementInputBase.SerializeCore(testElement, SaveContext.None);
+
+            var attributeValue = testElement.FirstChild.Attributes[0].Value;
+
+            Assert.AreEqual("20", attributeValue);
+        }
+
+        [Test]
+        public void UpdateValueCoreTest()
+        {
+            measurementInputBase.Value = 20;
+
+            //Update Value
+            var updateParam = new UpdateValueParams("Value", "1");
+            var result = measurementInputBase.UpdateValueCore(updateParam);
+            Assert.IsTrue(result);
+        }
+
+        [Test]
+        public void PreferenceSettings_PropertyChanged()
+        {
+            Open(@"UI\UnitsUITests.dyn");
+            ViewModel.HomeSpace.RunSettings.RunType = RunType.Manual;
+
+            Run();
+            var node = Model.CurrentWorkspace.NodeFromWorkspace<LengthFromString>("5705381c277c4a86bf6650aeda12a468");
+            Assert.IsFalse(node.IsModified);
+
+            Model.PreferenceSettings.NumberFormat = "0.0";
+
+            node = Model.CurrentWorkspace.NodeFromWorkspace<LengthFromString>("5705381c277c4a86bf6650aeda12a468");
```

Comment:
Why do you have to obtain the `node` again? It's already assigned in line 87?

Reply:
Echo that, copy pasted maybe?

Reply:
Those classes where already covered, missing only one constructor on each. Also, the constructors on those classes have an obsolete warning. Of the three implementations of `MeasurementInputBase`, `LengthFromString` is the only one without that warning.

https://github.com/DynamoDS/Dynamo/blob/f39159b46625049974a510a043543fababbf51f0/src/Libraries/UnitsUI/UnitsUI.cs#L328-L332

https://github.com/DynamoDS/Dynamo/blob/f39159b46625049974a510a043543fababbf51f0/src/Libraries/UnitsUI/UnitsUI.cs#L288-L292

Reply:
@Astul-Betizagasti I think to @aparajit-pratap 's point here, can we skip this line and use node you got from Line 87?

Reply:
Oh sure, for some reason I didn't see that comment before, I'll change that

Reply:
Oh sure, I didn't see that comment before for some reason, I will change that now and update